applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1087,11 +1009,23 @@ static void connect(const void *subscriber_id, uint8_t report_id)
 
 	rs->linked_rd = rd;
 
-	if (state.selected == subscriber) {
-		/* Route report data to report state. */
-		if (rd->linked_rs) {
+	if (rd->linked_rs) {
+		__ASSERT_NO_MSG(rd->linked_rs != rs);
+
+		int8_t cur_sub_prio = (rd->linked_rs->subscriber->is_usb) ?
```

Comment:
do we really need to add these macros? this would be yet another thing to track...

```
uint8_t cur_sub_prio = rd->linked_rs->subscriber->is_usb;
uint8_t new_sub_prio = subscriber->is_usb;

if (new_sub_prio > cur_sub_prio) {
   ///force unlink
}
```

I am not fond using negative priorities if it is not necessary (I expect you tried to replicate thread priorities).
It has disadvantage as it is quite easy to make a mistake when doing comparisons (as lower is now higher).

Reply:
I think that using priorities here could make it easier to expand the module in future if needed (e.g. swap priorities or add another subscribers). This could make hid_state code more generic too. In that case maybe we could leave it as it is ?
Regarding negative priorites, I wanted to replicate the thread priorities, but I could change this.

Reply:
Maybe we could add something like subscriber type instead of using `is_usb` boolean? That should not be a difficult change for us and it could let user to expand the hid_state module with new subscribers easily.

Reply:
I prefer simpler code. Priorities at this point are unnecessary if we need to add them for some reason this are few lines of code.

Reply:
Added one-liner function for priority to make code more readable.